File: Micro_Project/Spring Boot/Angular/Individual/EMS_DEPT/src/main/java/com/ems_dept/model/Employee.java
package com.ems_dept.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "emp")
public class Employee {

	@Id
	@Column(name = "eid")
	private int empId;

	@Column(name = "ename", length = 20)
	private String empName;

	@Column(name = "esalary")
	private int empSalary;

	@ManyToOne
	@JoinColumn(name = "dno")
	private Department dep;

	public int getEmpId() {
		return empId;
	}

	public void setEmpId(int empId) {
		this.empId = empId;
	}

	public String getEmpName() {
		return empName;
	}

	public void setEmpName(String empName) {
		this.empName = empName;
	}

	public int getEmpSalary() {
		return empSalary;
	}

	public void setEmpSalary(int empSalary) {
		this.empSalary = empSalary;
	}

	public Department getDep() {
		return dep;
	}

	public void setDep(Department dep) {
		this.dep = dep;
	}

}

File: Micro_Project/Spring Boot/React/Individual/ems-app/src/model/Employee.js
import Department from "./Department";

class Employee {
   empId;
   empName;
   empSalary;
   dep;
   constructor() {
      this.empId=0;
      this.empName="";
      this.empSalary=0;
      this.dep = new Department();

   }

    setEmpId(empId){
       this.empId=empId;
    }
    setEmpName(empName){
       this.empName=empName;
    }
    setEmpSalary(empSalary){
       this.empSalary=empSalary;
    }

    setDep(dep){
      this.dep=dep;
    }
}
export default Employee;
